### ISSUE TITLE
Gradle Lite manager does not respect some package rules

### ISSUE THREAD

**Issue by Xendar** [renovatebot/renovate#8772]
**What Renovate type, platform and version are you using?**

Self hosted Renovate, ran with Docker image `renovate/renovate:24.55.0`, plus reproducer repo: https://github.com/Xendar/renovate-bot-gradle-lite

**Describe the bug**

I have an hybrid Maven+Gradle private project for which I am using Renovate, with a configuration to exclude several upgrades using a various set of package rules.
Following the discussion on issue #8454 I switched from the Gradle manager to the Gradle Lite one. Doing so, Renovate created some PRs automatically which should be excluded according to my configuration. Interesting thing, it created a PR only for the Gradle dependency, not for the Maven one.

**Relevant debug logs**

I had a look at both self hosted and renovate app logs, and could not find anything relevant

**Have you created a minimal reproduction repository?**

I have created this project as a reproducer: https://github.com/Xendar/renovate-bot-gradle-lite
3 PRs have been opened by Renovate, using three different package rules that are not respected it seems, plus one branch created (resteasy major version update) but PR not yet opened

Renovate [configuration](https://github.com/Xendar/renovate-bot-gradle-lite/blob/master/renovate.json):
```json
{
  "extends": [
    "config:base"
  ],
  "gradle": {
    "enabled": false
  },
  "gradle-lite": {
    "enabled": true
  },
  "packageRules": [
    {
      "languages": [
        "java"
      ],
      "packageNames": [
        "org.jboss.resteasy:resteasy-bom",
        "org.jboss.resteasy:resteasy-cdi"
      ],
      "allowedVersions": "<= 3.1"
    },
    {
      "languages": [
        "java"
      ],
      "packageNames": [
        "org.eclipse.jgit:org.eclipse.jgit",
        "org.eclipse.jgit:org.eclipse.jgit.junit"
      ],
      "extends": [
        ":disableMajorUpdates"
      ]
    },
    {
      "languages": [
        "java"
      ],
      "packageNames": [
        "org.apache.maven:maven-model",
        "org.apache.maven:maven-model-builder"
      ],
      "enabled": false
    }
  ]
}
```

- [ ] This is a really small bug, it does not need a reproduction (think small typo)
- [x] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

Could it be that the manager does not pick the package rules? Or they are not picked up because the Gradle manager is explicitly disabled?

**Comment by rarkins** [comment]
Maybe gradle lite doesn't inherit from java. If you removed the languages filter or changed it to datasources=maven then it should work until the fix is made

**Comment by Xendar** [comment]
> Maybe gradle lite doesn't inherit from java. If you removed the languages filter or changed it to datasources=maven then it should work until the fix is made

That's interesting! I'll give it a try.

**Comment by Xendar** [comment]
> Maybe gradle lite doesn't inherit from java. If you removed the languages filter or changed it to datasources=maven then it should work until the fix is made

I just tried to replace the `language` by `matchDatasources` and it auto closed the faulty PRs indeed.
FYI I tried too to use `matchLanguages` instead of `languages` but it didn't change a thing

**Comment by rarkins** [comment]
Show your new package rules. Do not use languages.

**Comment by Xendar** [comment]
> Show your new package rules. Do not use languages.

Here it my last `renovate.json`:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "onboardingConfig": {
    "extends": [
      "config:base"
    ]
  },
  "hostRules": [
    {
      "hostType": "maven",
      "hostName": "https://my.company.artifactory/mvn-public/"
    }
  ],
  "gradle": {
    "enabled": false
  },
  "gradle-lite": {
    "enabled": true
  },
  "packageRules": [
    {
      "matchDatasources": [
        "maven"
      ],
      "packagePatterns": [
        "^com.company.some-package"
      ],
      "enabled": false
    },
    {
      "matchDatasources": [
        "maven"
      ],
      "packagePatterns": [
        "^com.company."
      ],
      "registryUrls": [
        "https://my.company.artifactory/mvn-public/"
      ]
    },
    {
      "matchDatasources": [
        "maven"
      ],
      "packageNames": [
        "com.company:some-package"
      ],
      "extends": [
        ":disableMajorUpdates"
      ]
    },
    {
      "matchDatasources": [
        "maven"
      ],
      "packageNames": [
        "org.jboss.resteasy:resteasy-bom",
        "org.jboss.resteasy:resteasy-cdi"
      ],
      "allowedVersions": "<= 3.1"
    },
    {
      "matchDatasources": [
        "maven"
      ],
      "packageNames": [
        "org.eclipse.jgit:org.eclipse.jgit",
        "org.eclipse.jgit:org.eclipse.jgit.junit"
      ],
      "extends": [
        ":disableMajorUpdates"
      ]
    },
    {
      "matchDatasources": [
        "maven"
      ],
      "packageNames": [
        "org.apache.maven:maven-model",
        "org.apache.maven:maven-model-builder"
      ],
      "enabled": false
    },
    {
      "matchDatasources": [
        "maven"
      ],
      "packageNames": [
        "org.eclipse.microprofile.health:microprofile-health-api",
        "org.apache.geronimo:geronimo-health"
      ],
      "extends": [
        ":disableMajorUpdates"
      ]
    },
    {
      "matchDatasources": [
        "maven"
      ],
      "packageRules": [
        {
          "packageNames": [
            "org.junit.jupiter:junit-jupiter-api",
            "org.junit.jupiter:junit-jupiter-params",
            "org.junit.jupiter:junit-jupiter-engine",
            "org.junit.vintage:junit-vintage-engine"
          ],
          "groupName": "junit dependencies"
        }
      ]
    },
    {
      "matchDatasources": [
        "maven"
      ],
      "packageRules": [
        {
          "packageNames": [
            "org.mockito:mockito-core",
            "org.mockito:mockito-junit-jupiter"
          ],
          "groupName": "mockito dependencies"
        }
      ]
    },
    {
      "matchDatasources": [
        "maven"
      ],
      "packageRules": [
        {
          "packageNames": [
            "io.prometheus:simpleclient",
            "io.prometheus:simpleclient_hotspot",
            "io.prometheus:simpleclient_servlet"
          ],
          "groupName": "prometheus dependencies"
        }
      ]
    },
    {
      "matchDatasources": [
        "maven"
      ],
      "packageRules": [
        {
          "packageNames": [
            "org.mapstruct:mapstruct",
            "org.mapstruct:mapstruct-processor"
          ],
          "groupName": "mapstruct dependencies"
        }
      ]
    }
  ]
}
```

**Comment by rarkins** [comment]
Sorry I may have misread when on mobile earlier. Did you mean that everything was *working* already, or was something still not working? I was asking for the updated rules in case it wasn't :)

**Comment by Xendar** [comment]
> Sorry I may have misread when on mobile earlier. Did you mean that everything was _working_ already, or was something still not working? I was asking for the updated rules in case it wasn't :)

Aaaah I understand better now :D
So switching from language to matchDatasources did work! And I was mentioning that I tried to use matchLanguages instead of languages (I did not notice that those options were renamed with this match prefix) thinking that it could do a difference, but it did not.
Now I understand that anyway it's better to use datasources over languages, right?
